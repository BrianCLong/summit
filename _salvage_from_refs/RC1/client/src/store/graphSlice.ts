import { createSlice, PayloadAction } from "@reduxjs/toolkit";
import { ElementDefinition } from "cytoscape";

export interface GraphDataState {
  nodes: ElementDefinition[];
  edges: ElementDefinition[];
}

const initialState: GraphDataState = {
  nodes: [],
  edges: [],
};

const graphSlice = createSlice({
  name: "graphData",
  initialState,
  reducers: {
    addNode(state, action: PayloadAction<ElementDefinition>) {
      state.nodes.push(action.payload);
    },
    addEdge(state, action: PayloadAction<ElementDefinition>) {
      state.edges.push(action.payload);
    },
    reset(state) {
      state.nodes = [];
      state.edges = [];
    },
  },
});

export const { addNode, addEdge, reset } = graphSlice.actions;
export default graphSlice.reducer;
