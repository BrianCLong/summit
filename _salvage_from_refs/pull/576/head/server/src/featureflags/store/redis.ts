export class RedisStore {
  async get(_k: string) {
    return undefined;
  }
  async set(_k: string, _v: any) {
    // no-op placeholder
  }
}
