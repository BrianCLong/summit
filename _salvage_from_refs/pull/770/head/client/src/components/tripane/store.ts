import { create } from 'zustand'

export type ThemeMode = 'light' | 'dark'

interface AppState {
  theme: ThemeMode
  history: ThemeMode[]
  future: ThemeMode[]
  toggleTheme: () => void
  undo: () => void
  redo: () => void
}

export const useAppStore = create<AppState>((set, get) => ({
  theme: 'light',
  history: [],
  future: [],
  toggleTheme: () => {
    const { theme, history } = get()
    const next: ThemeMode = theme === 'light' ? 'dark' : 'light'
    set({ theme: next, history: [...history, theme], future: [] })
  },
  undo: () => {
    const { history, future, theme } = get()
    if (history.length === 0) return
    const previous = history[history.length - 1]
    set({
      theme: previous,
      history: history.slice(0, -1),
      future: [theme, ...future],
    })
  },
  redo: () => {
    const { history, future, theme } = get()
    if (future.length === 0) return
    const next = future[0]
    set({
      theme: next,
      history: [...history, theme],
      future: future.slice(1),
    })
  },
}))
