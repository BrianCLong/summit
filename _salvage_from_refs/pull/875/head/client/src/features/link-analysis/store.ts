import { create } from 'zustand';

interface TimeRange {
  start: number;
  end: number;
}

interface AnalysisState {
  timeRange: TimeRange;
  activeQuery: string | null;
  pinned: Set<string>;
  setTimeRange: (range: TimeRange) => void;
  runQuery: (query: string) => void;
  togglePinned: (id: string) => void;
  clearPinned: () => void;
}

export const useAnalysisStore = create<AnalysisState>((set) => ({
  timeRange: { start: 0, end: 100 },
  activeQuery: null,
  pinned: new Set<string>(),
  setTimeRange: (timeRange) => set({ timeRange }),
  runQuery: (activeQuery) => set({ activeQuery }),
  togglePinned: (id) =>
    set((s) => {
      const next = new Set(s.pinned);
      next.has(id) ? next.delete(id) : next.add(id);
      return { pinned: next };
    }),
  clearPinned: () => set({ pinned: new Set<string>() }),
}));
