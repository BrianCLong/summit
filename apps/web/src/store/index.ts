import { configureStore } from "@reduxjs/toolkit";
import focusReducer from "../features/focusMode/focusSlice";
import viewSyncReducer from "../features/viewSync/viewSyncSlice";
import codexReducer from "../features/codex/codexSlice";
import historyReducer from "../features/history/historySlice";
import { historyMiddleware } from "../features/history/historyMiddleware";
import { enablePatches } from "immer";

enablePatches();

export const store = configureStore({
  reducer: {
    focus: focusReducer,
    viewSync: viewSyncReducer,
    codex: codexReducer,
    history: historyReducer,
  },
  middleware: (gDM) => gDM().concat(historyMiddleware),
});
export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
