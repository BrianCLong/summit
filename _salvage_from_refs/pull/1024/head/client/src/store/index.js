import { configureStore } from '@reduxjs/toolkit';
import authSlice from './slices/authSlice';
import graphSlice from './slices/graphSlice';
import uiSlice from './slices/uiSlice';
import graphInteraction from './slices/graphInteractionSlice';
import graphUISlice from './slices/graphUISlice'; // Import the new graphUISlice

export const store = configureStore({
  reducer: {
    auth: authSlice,
    graph: graphSlice,
    ui: uiSlice,
    graphInteraction,
    graphUI: graphUISlice, // Add the new graphUISlice
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: ['persist/PERSIST'],
      },
    }),
});
