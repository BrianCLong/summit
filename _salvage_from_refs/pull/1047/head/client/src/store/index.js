import { configureStore } from '@reduxjs/toolkit';
import authSlice from './slices/authSlice';
import graphSlice from './slices/graphSlice';
import uiSlice from './slices/uiSlice';

export const store = configureStore({
  reducer: {
    auth: authSlice,
    graph: graphSlice,
    ui: uiSlice,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: ['persist/PERSIST'],
      },
    }),
});
