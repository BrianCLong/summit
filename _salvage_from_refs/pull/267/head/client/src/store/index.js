import { configureStore } from '@reduxjs/toolkit';
import authSlice from './slices/authSlice';
import graphSlice from './slices/graphSlice';
import uiSlice from './slices/uiSlice';
import graphInteraction from './slices/graphInteractionSlice';

export const store = configureStore({
  reducer: {
    auth: authSlice,
    graph: graphSlice,
    ui: uiSlice,
    graphInteraction,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: ['persist/PERSIST'],
      },
    }),
});
