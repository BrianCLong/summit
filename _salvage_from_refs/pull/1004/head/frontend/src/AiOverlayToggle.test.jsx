import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import { store } from './store'
import AiOverlayToggle from './AiOverlayToggle'
import { describe, it, expect } from 'vitest'

describe('AiOverlayToggle', () => {
  it('renders consistently', () => {
    const { container } = render(
      <Provider store={store}>
        <AiOverlayToggle />
      </Provider>
    )
    expect(container).toMatchSnapshot()
  })
})
