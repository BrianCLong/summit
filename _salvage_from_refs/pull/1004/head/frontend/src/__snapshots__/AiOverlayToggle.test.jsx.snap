exports[`AiOverlayToggle renders consistently 1`] = `<div><button data-testid="ai-toggle">Show AI Insights</button></div>`;
